### ISSUE TITLE
StatusBar doesn't update with TabNavigator

### ISSUE THREAD

**Issue by jamesisaac** [react-community/react-navigation#342]
It's being suggested in #11 that StatusBar should be controlled via screen options, but I agree with @grabbou that this is unnecessary -- StatusBar already provides a nice component-based API, so no need for a new abstraction via navigationOptions.  The way it's described in the RN docs is:

> It is possible to have multiple StatusBar components mounted at the same time. The props will be merged in the order the StatusBar components were mounted.

This seems to work perfectly in react-navigation with StackRouter, but not TabRouter (changing tabs doesn't "unmount" the last StatusBar).

Specifically, it's the color (Android only) that I'm trying to change per screen, and I have the following config:

```js
TabNavigator({
  ...
}, {
  tabBarComponent: TabView.TabBarBottom,
  tabBarPosition: 'bottom',
  swipeEnabled: false,
  animationEnabled: false,
  lazyLoad: false,
})
```

**Comment by grabbou** [comment]
Yeah, I believe this will be solved by #51 when we got a way to listen whether screen is active. Then, we could either use an API in componentDidUpdate, or conditionally render `StatusBar` as required

**Comment by matt-oakes** [comment]
@jamesisaac @grabbou Is there a workaround for this currently?
